base/file.jl

```diff
@@ -230,23 +230,27 @@ julia> mkpath("intermediate_dir/actually_a_directory.txt") # creates two directo
 julia> isdir("intermediate_dir/actually_a_directory.txt")
 true
 
+julia> mkpath("my/test/dir/") # returns the original `path`
+"my/test/dir/"
 ```
 """
 function mkpath(path::AbstractString; mode::Integer = 0o777)
-    isdirpath(path) && (path = dirname(path))
     dir = dirname(path)
+    # stop recursion for `""`, `"/"`, or existed dir
     (path == dir || isdir(path)) && return path
     mkpath(dir, mode = checkmode(mode))
     try
-        mkdir(path, mode = mode)
+        # cases like `mkpath("x/")` will cause an error if `isdir(path)` is skipped
+        # the error will not be rethrowed, but it may be slower, and thus we avoid it in advance
```

Comment:
```suggestion
        # the error will not be rethrown, but it may be slower, and thus we avoid it in advance
```
This again to me reads more like a justification for why this PR does what it does, and less like a helpful explanation for someone reading the code down the line. I would just omit it.

or if you feel it has value, start more like this: "We could omit the isdir check. Then mkdir may throw an error, but we catch that anyway. However, it is more efficient to ..."

but this still seems artificial to me.

But perhaps I am missing something here?

Reply:
You entirely got it. I meant to leave a reminder for people who think the check is unnecessary